lib/Backend/amd64/LowererMDArch.cpp

```diff
@@ -1643,6 +1653,12 @@ LowererMDArch::GeneratePrologueStackProbe(IR::Instr *entryInstr, IntConstType fr
     // $done:
     //
 
+    if (this->m_func->IsTrueLeaf() &&
```

Comment:
I think we should be fine with IsLeaf here.  If a helper has an implicit call and calls JS code, it will probe again.